pkg/operator/ceph/cluster/controller.go

```diff
@@ -263,6 +323,61 @@ func (c *ClusterController) onAdd(obj interface{}) {
 // ************************************************************************************************
 // Update event functions
 // ************************************************************************************************
+func (c *ClusterController) onK8sNodeUpdate(oldObj, newObj interface{}) {
+	// skip forced resyncs
+	if reflect.DeepEqual(oldObj, newObj) {
+		return
+	}
+
+	// Checking for nodes where NoSchedule-Taint got removed
+	newNode, ok := newObj.(*v1.Node)
+	if !ok {
+		logger.Warningf("Expected NodeList but handler received %#v", newObj)
+	}
+
+	oldNode, ok := oldObj.(*v1.Node)
+	if !ok {
+		logger.Warningf("Expected NodeList but handler received %#v", oldObj)
+	}
+
+	for i := range newNode.Spec.Taints {
+		if newNode.Spec.Taints[i].Effect == "NoSchedule" {
+			logger.Debugf("Skipping -> Node %s is Unschedulable", newNode.Labels[apis.LabelHostname])
+			return
+		}
+	}
+	var found = false
```

Comment:
I think maybe the `// Checking for nodes where NoSchedule-Taint got removed` comment is more contextual here.